src/framework/radioButtons/RadioButtons.jsx

```diff
@@ -0,0 +1,101 @@
+
+import React, {
+  Component,
+} from 'react';
+import classNames from 'classnames';
+import PropTypes from 'prop-types';
+
+import { Text } from '../framework';
+
+export default class RadioButtons extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      selectedOption: this.props.selectedElement,
+    };
+
+    this.onSelectOption = this.onSelectOption.bind(this);
+  }
+
+  componentDidMount() {
+    if (this.state.selectedOption !== this.props.selectedElement) {
+      // https://reactjs.org/docs/react-component.html#componentdidmount
+      // eslint-disable-next-line react/no-did-mount-set-state
+      this.setState({
+        selectedOption: this.props.selectedElement
+      });
+    }
+  }
+
+  componentDidUpdate() {
+    if (this.state.selectedOption !== this.props.selectedElement) {
```

Comment:
You have 2 methods duplicated here.... this is a common symptom of not having a single source of truth....in this case for the selectedElement....If you ask me i would remove completely the state of the component and leave it up to the client to keep track of the selected element......

Reply:
This was done this way to keep consistency with other components in the UI Framework, such as `Dropdown`. But totally I agree with you in general. 👍